src/main/java/com/dreammaster/gthandler/multiAirFilter/GT_MetaTileEntity_AirFilter.java

```diff
@@ -176,10 +178,11 @@ public boolean checkRecipe(ItemStack aStack){
         mPollutionReduction=GT_Utility.safeInt((long)mPollutionReduction*mEfficiency/10000);
 
         GT_Pollution.addPollution(getBaseMetaTileEntity(), -mPollutionReduction);
-        if(mInventory[1].getItem() instanceof GT_MetaGenerated_Tool_01 &&
-                ((GT_MetaGenerated_Tool) mInventory[1].getItem()).getToolStats(mInventory[1]).getSpeedMultiplier()>0 &&
-                ((GT_MetaGenerated_Tool) mInventory[1].getItem()).getPrimaryMaterial(mInventory[1]).mToolSpeed>0 ) {
-            ((GT_MetaGenerated_Tool) mInventory[1].getItem()).doDamage(mInventory[1], 10L*(long) min(-mEUt / (float)damageFactorLow, Math.pow(-mEUt, damageFactorHigh)));
+        if (mInventory[1].getItem() instanceof GT_MetaGenerated_Tool_01 &&
+                ((GT_MetaGenerated_Tool) mInventory[1].getItem()).getToolStats(mInventory[1]).getSpeedMultiplier() > 0 &&
+                GT_MetaGenerated_Tool.getPrimaryMaterial(mInventory[1]).mToolSpeed > 0 &&
+                mInventory[1].getItemDamage() > 169 && mInventory[1].getItemDamage() < 180) {
```

Comment:
since this check is essentially duplicated and is not exactly readable, maybe extract it to a method?